internal/platform/platform_versioner.go

```diff
@@ -164,3 +166,21 @@ func (pv K8SBasedPlatformVersioner) CompareOpenShiftVersion(client Discoverer, c
 	curVersion := MapKnownVersion(info)
 	return curVersion.Compare(OpenShiftVersion{Version: version})
 }
+
+func (pv K8SBasedPlatformVersioner) ConsoleYAMLSamplesExists(client Discoverer, cfg *rest.Config) (bool, error) {
+	client, _, err := pv.DefaultArgs(client, cfg)
+	if err != nil {
+		log.Info("issue occurred while defaulting args for version lookup")
+		return false, err
+	}
+	apis, err := client.ServerResourcesForGroupVersion("console.openshift.io/v1")
+	if err != nil {
+		return false, errors.New("not getting openshift console api info: " + err.Error())
+	}
+	for _, api := range apis.APIResources {
+		if api.Name == "consoleyamlsamples" {
```

Comment:
We need this to be a reusable function that checks if any CRD of interest exists, rather than to hardcode a check for consoleyamlsamples

Reply:
I thought of something similar. Do you mean we have a func to allow user to pass in param. to query any crd?   This looks like go back to the constraint we encountered from the internal package: We don’t want to use string as param, so we define constants, the constants we defined has to be in the internal package, otherwise, cross referencing.

Reply:
Yes, the function would have to be in utils.go. Not sure what dependency issue you've encountered, but if create a test case in the PR, we can review and resolve it.

Reply:
This is the new method and checked in.  

CustomResourceExists(groupVersion string, apiResource string, cfg ...*rest.Config) (bool, error)

And I was thinking we may want to provide some kind of obj/struct, consts instead of two Stings as the first two params, so that user won't need to lookup somewhere for those available string values.  If this is the case, then we have to put these obj/struct or consts definitions in openshift pkg, but putting them in this pkg also means they can't be used inside platform pkg. for e.g. to create any test case.

Reply:
1- I know Jeremy created a whole framework for openshift/version detection in internal and then exposed simple methods in pkg. But that's not by design and it's not a pattern. If you are creating a function that does something, just create the function in pkg/util. There is no need to create a function that you call under internal, just for the sake of doing so.

2- Relatedly, I want to see a test case for your util function that shows me how the user will use your utility. But no, I don't think there is any good reason to create a `struc` for the user to provide the GVK, unless there is a standard and well-known one already.

Reply:
1 - I found it is good to call those func as they are testable.  
[I've this not testable version](https://github.com/myeung18/operator-utils/blob/7108decf77c4e3d6e8d621a4c5722e3cdc261c61/pkg/utils/openshift/utils.go#L64-L94)
If there is no reason to have those func in internal, why don't we copy them all to utils pkg? 
As this will make the OpenShiftVersion public and semantic compatible, which is needed in other app. like kie-cloud. 
   
2. added the test cases

Reply:
1- There is a reason for that stuff to be in internal. I'm just saying it doesn't apply to everything you do, so whenever you are adding new functionality, don't think you have to provide the implementation under internal and some kind of facade in pkg.
2- I don't see a unit test that shows me how to use your library.

Reply:
You mean e2e test case this method utils.CustomResourceExists(....)? If yes, I can create few.  But I can't inject the mock for it, is it?
so the test cases are in platform.TestCustomResourceExists(t *testing.T);

Reply:
Let's start over again. We are looking for a simple function that tells us if a CRD (not a CR) exists in the cluster. CRD is the schema, so presumably you would know by checking the GVK (group + version + kind) in there. This would ideally be a single function implementation in kubernetes/utils.go (I imagine it's not OpenShift proprietary). It would also be a standalone PR that you can create, test, and open in a matter of a couple of hours, and nothing too complicated.

Reply:
do you mean sth like [this](https://github.com/myeung18/operator-utils/blob/f67826a2df210f354c95da51f0f35fb89c9a5e79/pkg/utils/openshift/utils.go#L67-L83).
and to make it testable, I create [this](https://github.com/myeung18/operator-utils/blob/7108decf77c4e3d6e8d621a4c5722e3cdc261c61/pkg/utils/openshift/utils.go#L98-L150
):

Reply:
No.

I want to see a single function in utils.go. It should be named properly as per the above. It would take a GVK and say if that CRD is installed or not.

There should be a unit test for this function in utils_test.go

Reply:
@bmozaffa @ccustine

new func add and I am still working on the mock.